src/utils/hashlink.ts

```diff
@@ -0,0 +1,118 @@
+   * Encodes a buffer, with optional metadata, into a hashlink
+   *
+   * @param {Buffer} buffer the buffer to encode into a hashlink
+   * @param {string} hashName the name of the hashing algorithm 'sha2-256'
+   * @param {BaseName} baseName the name of the base encoding algorithm 'base58btc'
+   * @param {Metadata} metadata the optional metadata in the hashlink
+   */
+  public static encode(buffer: Buffer, hashName: 'sha2-256', baseName: BaseName = 'base58btc', metadata?: Metadata) {
+    const checksum = this.encodeMultihashEncoder(buffer, hashName, baseName)
+    const mbMetadata = metadata ? this.encodeMetadata(metadata, baseName) : null
+    return mbMetadata ? `hl:${checksum}:${mbMetadata}` : `hl:${checksum}`
+  }
+
+  /**
+   *
+   * Decodes a hashlink into HashlinkData object
+   *
+   * @param {string} hashlink the hashlink that needs decoding
+   */
+  public static decode(hashlink: string): HashlinkData {
+    const hashlinkList = hashlink.split(':')
+    if (this.validate(hashlink)) {
+      const checksum = hashlinkList[1]
+      const metadata = hashlinkList[2] ? this.decodeMetadata(hashlinkList[2]) : null
+
+      return metadata ? { checksum, metadata } : { checksum }
+    } else {
+      throw new Error(`Hashlink, ${hashlink}, is invalid`)
+    }
+  }
+
+  public static validate(hashlink: string): boolean {
+    const hashlinkList = hashlink.split(':')
+    const validMultibase = MultibaseEncoder.validate(hashlinkList[1])
+    const { data } = MultibaseEncoder.decode(hashlinkList[1])
+    const validMultihash = MultihashEncoder.validate(data)
+    return validMultibase && validMultihash ? true : false
+  }
+
+  private static encodeMultihashEncoder(buffer: Buffer, hashName: 'sha2-256', baseName: BaseName): string {
+    // TODO: Support more hashing algorithms
+    const hash = new Uint8Array(sha256.array(buffer))
+    const mh = MultihashEncoder.encode(hash, hashName)
+    const mb = MultibaseEncoder.encode(mh, baseName)
+    return BufferEncoder.toUtf8String(mb)
+  }
+
+  private static encodeMetadata(metadata: Metadata, baseName: BaseName): string {
+    const metadataMap = new Map()
+
+    for (const key of Object.keys(metadata)) {
+      switch (key) {
+        case 'urls':
+          metadataMap.set(URLS, metadata.urls)
+          break
+        case 'contentType':
+          metadataMap.set(CONTENTTYPE, metadata.contentType)
+          break
+        default:
+          throw new Error(`Metadata, ${metadata}, is invalid`)
+      }
+    }
+
+    const cborData = cbor.encode(metadataMap)
+
+    const multibaseMetadata = MultibaseEncoder.encode(cborData, baseName)
+
+    return BufferEncoder.toUtf8String(multibaseMetadata)
+  }
+
+  private static decodeMetadata(mb: string): Metadata {
+    const obj = { urls: [], contentType: '' }
+    const { data } = MultibaseEncoder.decode(mb)
+    try {
+      // eslint-disable-next-line @typescript-eslint/no-explicit-any
+      const cborData: Map<number, any> = cbor.decode(data)
```

Comment:
```suggestion
      const cborData: Map<number, unknown> = cbor.decode(data)
```

Reply:
Unknown does not work in this situation, because of:

`Type 'unknown' is not assignable to type 'string[]'.ts(2322)`

When assigning the url list to the url list in the object.